### ISSUE TITLE
Adding a waypoint doesn't lock in the path from the start to the waypoint

### ISSUE THREAD

**Issue by sumpfralle** [triplea-game/triplea#4530]
### Engine version
1.10.13507

### My Operating System
Debian sid

### Map name
Total World War 1941 beta

### Can you describe how to trigger the error? (eg: what sequence of actions will recreate it?)
Start with a train in Berlin, select a waypoint (via CTRL-T) in Pskov (distance: 5) and try to select the next step (any). After pressing CTRL-T over Pskov, every valid next step is forbidden.

### Do you have the exact error text? Please copy/paste if so
The error message at the bottom of the window says *Invalid move, only start and end can be land when route has water*. This message does not fit to the situation.

Please see the attached screenshot.
The example savegame is the same as the one supplied to #4528.

## EDIT
- The issue here is that the path from the start to the waypoint can change after setting the waypoint and become an invalid path

**Comment by ron-murhammer** [comment]
@sumpfralle Can you retest this with the latest pre-release and if you still see an issue then post a screenshot?

**Comment by sumpfralle** [comment]
It works - thank you!

**Comment by sumpfralle** [comment]
My bad - I mixed it up with my other bug report. Indeed the problem exists in the current release (1.10.13574).

![triplea-waypoint-invalidates-move](https://user-images.githubusercontent.com/1474141/50743884-40154c80-121d-11e9-99cb-4d0f67547f6e.png)

**Comment by ron-murhammer** [comment]
@sumpfralle Yeah, this is a current limitation of way points which is it may choose a different path then you initial see from your start to the way point. In this case you can see it tries to go over the Baltic Sea. If you add a waypoint in between Berlin and Pskov then it should work.

**Comment by DanVanAtta** [comment]
Moved to: https://github.com/triplea-game/triplea/wiki/UX-Improvements

**Comment by VictoryFirst2** [comment]
Tested this with the newest version and I am not able to reproduce this. It looks like it has been fixed.
